MetadataMigration/src/main/java/org/opensearch/migrations/cli/Items.java

```diff
@@ -31,25 +34,73 @@ public String asCliOutput() {
         } else {
             sb.append("Migrated Items:" + System.lineSeparator());
         }
-        sb.append(Format.indentToLevel(1) + "Index Templates:" + System.lineSeparator());
-        sb.append(Format.indentToLevel(2) + getPrintableList(getIndexTemplates()) + System.lineSeparator());
-        sb.append(System.lineSeparator());
-        sb.append(Format.indentToLevel(1) + "Component Templates:" + System.lineSeparator());
-        sb.append(Format.indentToLevel(2) +getPrintableList(getComponentTemplates()) + System.lineSeparator());
-        sb.append(System.lineSeparator());
-        sb.append(Format.indentToLevel(1) + "Indexes:" + System.lineSeparator());
-        sb.append(Format.indentToLevel(2) + getPrintableList(getIndexes()) + System.lineSeparator());
-        sb.append(System.lineSeparator());
-        sb.append(Format.indentToLevel(1) + "Aliases:" + System.lineSeparator());
-        sb.append(Format.indentToLevel(2) +getPrintableList(getAliases()) + System.lineSeparator());
+        appendSection(sb, "Index Templates", getIndexTemplates());
+        appendSection(sb, "Component Templates", getComponentTemplates());
+        appendSection(sb, "Indexes", getIndexes());
+        appendSection(sb, "Aliases", getAliases());
+
+        return sb.toString();
+    }
+
+    private void appendSection(StringBuilder sb, String sectionTitle, List<CreationResult> items) {
+        sb.append(Format.indentToLevel(1))
+          .append(sectionTitle)
+          .append(":")
+          .append(System.lineSeparator());
+
+        if (items.isEmpty()) {
+            sb.append(Format.indentToLevel(2))
+                .append(NONE_FOUND_MARKER)
+                .append(System.lineSeparator());
+          } else {
+            appendItems(sb, items);
+            appendFailures(sb, items);
+        }
         sb.append(System.lineSeparator());
+    }
+
+    private void appendItems(StringBuilder sb, List<CreationResult> items) {
+        var successfulItems = items.stream()
+            .filter(r -> r.wasSuccessful())
+            .map(r -> r.getName())
+            .collect(Collectors.toList());
+
+        if (!successfulItems.isEmpty()) {
+            sb.append(Format.indentToLevel(2))
```

Comment:
nit: would prefer `2` to be a parameter to this function and appendFailures to be less errorProne if extending

Reply:
I'll leave this as is for now, but if we invest more in the CLI experience in java this could use some more thought